lib/services/ngsi/ngsiService.js

```diff
@@ -106,6 +107,80 @@ function generateNGSIOperationHandler(operationName, entityName, typeInformation
     };
 }
 
+/**
+ * Generate an operation handler for NGSIv2-based operations (query and update). The handler takes care of identifiying
+ * the errors and calling the appropriate callback with a success or a failure depending on how the operation ended.
+ *
+ * Most of the parameters are passed for debugging purposes mainly.
+ *
+ * @param {String} operationName        Name of the NGSI operation being performed.
+ * @param {String} entityName           Name of the entity that was the target of the operation.
+ * @param {Object} typeInformation      Information about the device the entity represents.
+ * @param {String} token                Security token used to access the entity.
+ * @param {Object} options              Object holding all the information about the HTTP request.
+
+ * @return {Function}                  The generated handler.
+ */
+function generateNGSI2OperationHandler(operationName, entityName, typeInformation, token, options, callback) {
+    return function(error, response, body) {
+        if (error) {
+            logger.error(context, 'Error found executing ' + operationName + ' action in Context Broker: %s', error);
+
+            alarms.raise(constants.ORION_ALARM, error);
+            callback(error);
+        } else if (body && body.orionError) {
+            logger.debug(context, 'Orion error found executing ' + operationName + ' action in Context Broker: %j',
+                body.orionError);
+
+            callback(new errors.BadRequest(body.orionError.details));
+        } else if (response &&
+            operationName === 'update' && (response.statusCode === 204)) {
+            logger.debug(context,
+                'Received the following request from the CB:\n\n%s\n\n', JSON.stringify(body, null, 4));
+                logger.debug(context, 'Value updated successfully');
+                alarms.release(constants.ORION_ALARM);
+                callback(null, body);
+        } else if (response &&
+            operationName === 'query' && body !== undefined &&
```

Comment:
It is a bit weird to consider a 204 Not Content in a CB query operation, because any query should return a body (even and empty one, such as `[]`). 

Maybe you should check for the `operationName ===  'query' && response.statusCode === 204` separately and log an warn/error in that case.

Reply:
Fixed in 51f4e00371143d068e3f9e3549141beec73f25ac